test/tests/restart/restart_subapp_not_master/two_step_solve_sub_restart.i

```diff
@@ -0,0 +1,82 @@
+[Mesh]
+  file = two_step_solve_master_cp/0002_mesh.cpr
+[]
+
+[Problem]
+  restart_file_base = two_step_solve_master_cp/LATEST
+  allow_sub_app_restart = true
```

Comment:
Oh this is way more complicated than that. We do have a few restart tests that "appear" to work with recover (even though they don't actually https://github.com/idaholab/moose/issues/5877). Some of this restart/recover logic gets very complicated in that scenario. I'm having a really difficult time pulling that flag out and getting all the right paths to work right now. 

It may be worth disallowing restart + recover altogether including disabling the currently broken tests instead of allowing them to pass incorrectly as I pull that flag out.